### PR TITLE
Update networking ingress types for Ingress Backend Service Port name

### LINKED ISSUE #97107: Ingress ServicePort and Service Port Name are validated with different schemas
**What happened**: Tried to create an ingress with a backend serviceport pointing to a service with a port name with length > 15 chars that is valid from the service point of view. But failed because the validation for Ingress does not match the one for service.

Service Validation (uses DNS Label):
- https://github.com/kubernetes/kubernetes/blob/e1c617a88ec286f5f6cb2589d6ac562d095e1068/pkg/apis/core/validation/validation.go#L4359)

Ingress Service Port Validation (uses IANA format like for container ports): 
-  https://github.com/kubernetes/kubernetes/blob/e1c617a88ec286f5f6cb2589d6ac562d095e1068/pkg/apis/networking/validation/validation.go#L464
- https://github.com/kubernetes/kubernetes/blob/e1c617a88ec286f5f6cb2589d6ac562d095e1068/staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go#L327

**What you expected to happen**: Service port validation should match Ingress ServicePort validation

**How to reproduce it (as minimally and precisely as possible)**:

```yaml
# Service
apiVersion: v1
kind: Service
metadata:
  name: nginx-vault-http
spec:
  type: NodePort
  selector:
    app.kubernetes.io/name: "nginx-vault"
  ports:
  - name: http-nginx-vault
    port: 80
    protocol: TCP
    targetPort: http

# Ingress
apiVersion: networking.k8s.io/v1beta1
kind: Ingress
metadata:
  name: nginx-vault-http
  annotations:
    kubernetes.io/ingress.class: "cluster-haproxy"
spec:
  rules:
  - host: www.my-nginx-vault.com
    http:
      paths:
      - path: "/"
        backend:
          serviceName: "nginx-vault-http"
          servicePort: http-nginx-vault
```
**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
Client Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.4", GitCommit:"d360454c9bcd1634cf4cc52d1867af5491dc9c5f", GitTreeState:"clean", BuildDate:"2020-11-16T17:18:52Z", GoVersion:"go1.15.4", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.10", GitCommit:"62876fc6d93e891aa7fbe19771e6a6c03773b0f7", GitTreeState:"clean", BuildDate:"2020-10-15T01:43:56Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}

- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

### PR DESCRIPTION
**Pull request by filintod**
Add minor documentation explaining that the ServiceBackendPort Name should follow the IANA RFC6335 format

**What type of PR is this?**

/kind documentation

**What this PR does / why we need it**:

The Service Port names follow a different validation to the ingress service port names but it is not expressed in the current doc.

**Which issue(s) this PR fixes**:
Fixes #97107

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/apis/networking/types.go [hunk 1]
```diff
@@ -490,7 +490,7 @@ type IngressServiceBackend struct {
 
 // ServiceBackendPort is the service port being referenced.
 type ServiceBackendPort struct {
-	// Name is the name of the port on the Service.
+	// Name is the name of the port on the Service. This must be an IANA_SVC_NAME (following RFC6335).
```
Comment (aojea) [inline pkg/apis/networking/types.go:493]: nit, it looks better if we add it in the next line

Comment (filintod) [reply pkg/apis/networking/types.go]: done

**Review by pacoxu — APPROVED** [review]

**Review by thockin — COMMENTED** [review]
Thanks!

/lgtm
/approve

**Pull request merged**